### ISSUE TITLE
Metal: failed assertion `New volume: 6 should match old volume: 24` for einsum "ijk,kji->kij"

### ISSUE THREAD

**Issue by dlwh** [google/jax#20115]
### Description

distinct failure mode from #20114 

```python
>>> import jax.numpy as jnp
>>> a = jnp.ones((2,3,4))
Platform 'METAL' is experimental and not all JAX functionality may be correctly supported!
2024-03-06 22:50:31.964310: W pjrt_plugin/src/mps_client.cc:563] WARNING: JAX Apple GPU support is experimental and not all JAX functionality is correctly supported!
Metal device set to: Apple M1 Pro

systemMemory: 32.00 GB
maxCacheSize: 10.67 GB

>>> b = jnp.ones((4,3,2))
>>>  jnp.einsum("ijk,kji->kij", a, b)
  File "<stdin>", line 1
    jnp.einsum("ijk,kji->kij", a, b)
IndentationError: unexpected indent
>>> jnp.einsum("ijk,kji->kij", a, b)
/AppleInternal/Library/BuildRoots/4e1473ee-9f66-11ee-8daf-cedaeb4cabe2/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSNDArray/Kernels/MPSNDArrayIdentity.mm:813: failed assertion `New volume: 6 should match old volume: 24 [reshapeWithCommandBuffer] MPSNDArrayIdentity.'
Abort trap: 6
```

### System info (python version, jaxlib version, accelerator, etc.)

```
python -c "import jax; jax.print_environment_info()"
Platform 'METAL' is experimental and not all JAX functionality may be correctly supported!
2024-03-06 22:52:59.325589: W pjrt_plugin/src/mps_client.cc:563] WARNING: JAX Apple GPU support is experimental and not all JAX functionality is correctly supported!
Metal device set to: Apple M1 Pro

systemMemory: 32.00 GB
maxCacheSize: 10.67 GB

jax:    0.4.20
jaxlib: 0.4.20
numpy:  1.26.4
python: 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:35:25) [Clang 16.0.6 ]
jax.devices (1 total, 1 local): [METAL(id=0)]
process_count: 1
```

**Comment by shuhand0** [comment]
The issue will be fixed in the upcoming release.

**Comment by shuhand0** [comment]
Fixed in jax-metal 0.0.6.
